cdisc_rules_engine/models/validation_error_container.py

```diff
@@ -1,32 +1,35 @@
 from typing import List, Union
-
-from cdisc_rules_engine.enums.execution_status import ExecutionStatus
+from dataclasses import dataclass
 from cdisc_rules_engine.utilities.utils import get_execution_status
 
 from .base_validation_entity import BaseValidationEntity
 from .failed_validation_entity import FailedValidationEntity
 from .validation_error_entity import ValidationErrorEntity
 
 
+@dataclass
 class ValidationErrorContainer(BaseValidationEntity):
+    dataset: str | None = None
+    domain: str | None = None
+    targets: List[str] = None
+    errors: List[Union[ValidationErrorEntity, FailedValidationEntity]] = None
+    message: str | None = None
+
     def __init__(self, **params):
-        self.dataset: str = params.get("dataset")
-        self.domain: str = params.get("domain")
-        self.targets: List[str] = params.get("targets", [])
-        self.errors: List[Union[ValidationErrorEntity, FailedValidationEntity]] = (
-            params.get("errors", [])
-        )
-        self.message: str = params.get("message")
-        self.status: ExecutionStatus = params.get("status") or get_execution_status(
-            self.errors
-        )
+        super().__init__()
+        self.dataset = params.get("dataset")
+        self.domain = params.get("domain")
+        self.targets = params.get("targets", [])
+        self.errors = params.get("errors", [])
+        self.message = params.get("message")
+        self.status = params.get("status") or get_execution_status(self.errors)
 
-    def to_representation(self) -> dict:
+    def as_dict(self) -> dict:
```

Comment:
don't define `as_dict` explicitly if not needed. use `repr=false` where necessary. Make this update to all relevant files.